x-pack/solutions/observability/plugins/observability_ai_assistant_app/server/rule_connector/index.ts

```diff
@@ -177,6 +195,52 @@ async function executor(
     await resources.plugins.actions.start()
   ).getActionsClientWithRequest(request);
 
+  await Promise.all(
+    params.prompts.map((prompt) =>
+      executeAlertsChatCompletion(
+        resources,
+        prompt,
+        params,
+        alertDetailsContextService,
+        client,
+        functionClient,
+        actionsClient,
+        execOptions.logger
+      )
+    )
+  );
+
+  return { actionId: execOptions.actionId, status: 'ok' };
+}
+
+async function executeAlertsChatCompletion(
+  resources: ObservabilityAIAssistantRouteHandlerResources,
+  prompt: { statuses: string[]; message: string },
+  params: ConnectorParamsType,
+  alertDetailsContextService: AlertDetailsContextualInsightsService,
+  client: ObservabilityAIAssistantClient,
+  functionClient: ChatFunctionClient,
+  actionsClient: PublicMethodsOf<ActionsClient>,
+  logger: Logger
+): Promise<void> {
+  const alerts = {
+    new: [...(params.alerts?.new || [])],
+    recovered: [...(params.alerts?.recovered || [])],
+  };
+
+  if (ALERT_STATUSES.some((status) => prompt.statuses.includes(status))) {
```

Comment:
Why do you need to iterate through `ALERT_STATUSES`? Won't `prompt.statuses` always be `AlertStatus[]` (whatever that type is called)?   Could this simply be a check on whether it's empty?
```suggestion
  if (prompt.statuses.length > 0) {
```